mpp/include/mpp_do_update_ad.fh

```diff
@@ -122,53 +123,41 @@
                call mpp_error(FATAL, "mpp_do_update: mismatch on send and recv size")
             endif
          enddo
-         call mpp_sync_self()
          write(outunit,*)"NOTE from mpp_do_update: message sizes are matched between send and recv for domain " &
                           //trim(domain%name)
-         deallocate(msg1, msg2)
+         deallocate(msg1, msg2, msg3)
       endif
 
-      !recv
+      ! checkpoint buffer recv size and send pos and msgsize
+      ! forward recv info
```

Comment:
It would be helpful to explain the purpose of this loop.

Reply:
added more comment lines